src/js/profiling/convertHermesProfile.ts

```diff
@@ -0,0 +1,94 @@
+import type { StackId, ThreadCpuFrame,ThreadCpuSample, ThreadCpuStack, ThreadId } from '@sentry/types';
+import { logger } from '@sentry/utils';
+
+import type * as Hermes from './hermes';
+import { parseHermesStackFrameName } from './hermes';
+import type { RawThreadCpuProfile } from './types';
+
+const UNKNOWN_FRAME_FUNCTION_NAME = '<unknown>';
+const UNKNOWN_STACK_ID = -1;
+const JS_THREAD_NAME = 'JavaScriptThread';
+const JS_THREAD_PRIORITY = 1;
+
+/**
+ * Converts a Hermes profile to a Sentry profile.
```

Comment:
Can you describe roughly how the conversion is done? There is a lot happening in this function that might not be immediately obvious to the reader